src/pkg/quota/dao/dao.go

```diff
@@ -171,47 +170,57 @@ func (d *dao) GetByRef(ctx context.Context, reference, referenceID string) (*mod
 	return toQuota(quota, usage), nil
 }
 
-func (d *dao) GetByRefForUpdate(ctx context.Context, reference, referenceID string) (*models.Quota, error) {
+func (d *dao) Update(ctx context.Context, quota *models.Quota) error {
 	o, err := orm.FromContext(ctx)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	quota := &Quota{Reference: reference, ReferenceID: referenceID}
-	if err := o.ReadForUpdate(quota, "reference", "reference_id"); err != nil {
-		return nil, orm.WrapNotFoundError(err, "quota not found for (%s, %s)", reference, referenceID)
+	if quota.UsedChanged && quota.HardChanged {
+		return errors.New("not support change both hard and used of the quota")
 	}
 
-	usage := &QuotaUsage{Reference: reference, ReferenceID: referenceID}
-	if err := o.ReadForUpdate(usage, "reference", "reference_id"); err != nil {
-		return nil, orm.WrapNotFoundError(err, "quota usage not found for (%s, %s)", reference, referenceID)
+	if !quota.UsedChanged && !quota.HardChanged {
+		return nil
 	}
 
-	return toQuota(quota, usage), nil
-}
+	var (
+		sql    string
+		params []interface{}
+	)
 
-func (d *dao) Update(ctx context.Context, quota *models.Quota) error {
-	o, err := orm.FromContext(ctx)
+	if quota.UsedChanged {
+		sql = "UPDATE quota_usage SET used = ?, update_time = ?, version = ? WHERE id = ? AND version = ?"
```

Comment:
In case someone use the updated version in the following logic. Should not we refresh the it for model quota after it's updated?

Reply:
The callers need to get the quota again themself when they want to update the quota again after they updated the quota variable, otherwise, the second updating will always fail.